### ISSUE TITLE
Build crash because of a shader compilation issue

### ISSUE THREAD

**Issue by AnKO314** [wave-harmonic/crest#819]
**Describe the bug**
Build crash because of a shader error

**Log**
Error building Player: Shader error in 'Hidden/Crest/Simulation/Update Shadow': undeclared identifier 'unity_StereoMatrixVP' at line 93 (on d3d11)

**Versions**
Unity 2020.3.0f1
HDRP 10.3.2

**Platform**
 - Stand alone 64

**Additional context**
The missing matrix is referenced at this line : 

```shader
// unity_MatrixVP comes from **EditorShaderVariables**.hlsl. Not the same as UNITY_MATRIX_VP.
output.positionCS = mul(**unity_MatrixVP**, float4(input.positionOS, 1.0));
```

EditorShadervariables.hlsl content : 

```shader
#ifndef EDITOR_SHADER_VARAIBLES
#define E DITOR_SHADER_VARAIBLES

// ================================
//     PER FRAME CONSTANTS
// ================================
#if defined(USING_STEREO_MATRICES)
    #define glstate_matrix_projection unity_StereoMatrixP[unity_StereoEyeIndex]
    #define unity_MatrixV unity_StereoMatrixV[unity_StereoEyeIndex]
    #define unity_MatrixInvV unity_StereoMatrixInvV[unity_StereoEyeIndex]
    #define **unity_MatrixVP** unity_StereoMatrixVP[unity_StereoEyeIndex]

    #define unity_CameraProjection unity_StereoCameraProjection[unity_StereoEyeIndex]
    #define unity_CameraInvProjection unity_StereoCameraInvProjection[unity_StereoEyeIndex]
    #define unity_WorldToCamera unity_StereoWorldToCamera[unity_StereoEyeIndex]
    #define unity_CameraToWorld unity_StereoCameraToWorld[unity_StereoEyeIndex]
#else
    #if !defined(USING_STEREO_MATRICES)
        float4x4 glstate_matrix_projection;
        float4x4 unity_MatrixV;
        float4x4 unity_MatrixInvV;
        float4x4 **unity_MatrixVP**;
        float4 unity_StereoScaleOffset;
    #endif
#endif

#endif // EDITOR_SHADER_VARAIBLES
```

I tried to add a float4x4 unity_MatrixVP; in the faulty shader but unsurprinsigly, it reported an already declared error.
I don't know why the EditorShadervariables wouldn't be included in the build because it is located in the runtime folder of the hdrp package.

**Comment by daleeidd** [comment]
Thanks.

I reported this to Unity a while ago but forgot to reply them. I don't think it will go anywhere as the matrix is considered legacy according to comments.

I am just going to have to pass our own matrix in I think. I will look into having this done before the next version.

**Comment by daleeidd** [comment]
Found the problem. I was relying on an old check to determine if VR/XR is running which doesn't work for SRP.

In the file *Crest\Crest\Scripts\LodData\Shadows\SampleShadows.cs* change the two instances of *camera.stereoEnabled* to *XRGraphics.enabled*.

Fix will be out for 4.11.

**Comment by AnKO314** [comment]
> 
> 
> Found the problem. I was relying on an old check to determine if VR/XR is running which doesn't work for SRP.
> 
> In the file _Crest\Crest\Scripts\LodData\Shadows\SampleShadows.cs_ change the two instances of _camera.stereoEnabled_ to _XRGraphics.enabled_.
> 
> Fix will be out for 4.11.

Thank you ! I will try this soon !

**Comment by daleeidd** [comment]
Closing as 4.11 is out. Thank you for reporting the issue.
